src/devicecontroller/NoOpDeviceController.ts

```diff
@@ -70,6 +71,21 @@ export default class NoOpDeviceController
     return null;
   }
 }
+export class NoOpDeviceControllerWithEventController extends NoOpDeviceController {
+  private eventController: EventController;
+  constructor(eventController?: EventController) {
+    super();
+    this.eventController = eventController;
+  }
+
+  getEventController(): EventController {
```

Comment:
Do we need to add the `undefined` to the type signature here? `this.eventController` might be `undefined` here if I'm not wrong

```ts
getEventController(): EventController | undefined {};
```

Reply:
Ah yeah, before I had the constructor require a eventController so that wasn't a option, but yeah it should have undefined now.